nautobot/extras/models/customfields.py

```diff
@@ -398,29 +392,19 @@ class CustomField(BaseModel, ChangeLoggedModel, NotesMixin):
     ]
 
     class Meta:
-        ordering = ["weight", "name"]
+        ordering = ["weight", "label"]
 
     def __str__(self):
-        return self.label or self.name.replace("_", " ").capitalize()
+        return self.label
 
     def _fixup_empty_fields(self):
         """Handle the case when a new instance is created and some fields are left blank."""
         if self.present_in_database:
             return
 
-        # 2.0 TODO: this is to handle the UI case where `name` is no longer a directly configured form.
-        # Once `name` is no longer a model field, we can remove this.
-        if self.slug and not self.name:
-            self.name = self.slug
-
-        # 2.0 TODO: this is to fixup existing ORM usage when caller specifies a name but not a label;
-        # in 2.0 we should make `label` a mandatory field when getting rid of `name`.
-        if self.name and not self.label:
-            self.label = self.name
-
-        # This is to fix up existing ORM usage when caller doesn't specify a slug since it wasn't a field before.
-        if not self.slug:
-            self.slug = slugify_dashes_to_underscores(self.label or self.name)
+        # This is to fix up existing ORM usage when caller doesn't specify a key since it wasn't a field before.
+        if not self.key:
+            self.key = slugify_dashes_to_underscores(self.label)
```

Comment:
Is this/was this required with `key` being an `AutoSlugField`? I commented these lines out and it seemed to work fine.

```py
>>> cf = CustomField(label="this is my custom field")
>>> cf.save()
>>> cf.refresh_from_db()
>>> cf.key
'this_is_my_custom_field'
```

Reply:
Good catch. That's built-in logic in django-extensions AutoSlugField IIRC, we might need to override it in our AutoSlugField subclass.

Reply:
This behavior seems to go away after I restored `_fixup_empty_fields()`. Now this is the output
`DETAIL:  Key (key)=(this_is_my_custom_field) already exists.`

Reply:
I still think this can go away. Can you try setting `separator="_"` on the `AutoSlugField`?

Reply:
```
cf = CustomField(type=CustomFieldTypeChoices.TYPE_TEXT, label="My Field", required=False)
cf.validated_save()
```
It did fix the trailing index problem, but when I did the above operation in the ORM, it failed without `_fixup_empty_fields()`.

```
django.core.exceptions.ValidationError: {'key': ['This field cannot be blank.'], "CustomField's specific key attribute": ['This key is not Python/GraphQL safe. Please do not start the key with a digit and do not use hyphens or whitespace']}
```

Reply:
That's because `blank=False` is set on the field. This is probably the reason the default for `AutoSlugField` is `blank=True`

Reply:
Maybe just add a TODO note or open an issue to look into whether the AutoSlugField is being used incorrectly?